File: agent-driven-architecture-main/examples/kotlin/src/test/fixtures/konsist/compliant/C7/blocks/triage/Tools.kt
// ALLOW-TEST C7, the other half — the ONE place a block's transport is produced.
// One registration, two maps: this file supplies the pure `run` that mints the
// ToolResult, and the `sign` entry that turns it into a Command — the two legal
// constructions, in the one file entitled to both.
//
// Note how `sign` never names Signature: the lambda's parameters are inferred
// from the Verb seam, which is what lets C4(b) keep the stamp unnameable in a
// tool file while the sign entry still carries it through.
package adr.blocks.triage

import adr.contract.TriageCommand
import adr.contract.TriageResult
import adr.spine.pure.ToolName
import adr.spine.pure.Verb

val SET_PRIORITY = ToolName("setPriority")

class TriageTools {

    fun verbs(): List<Verb<*, *, *>> = listOf(
        Verb.Reversible(
            name = SET_PRIORITY,
            describe = "Set a support ticket's priority.",
            run = { input, _ -> TriageResult.SetPriority(SET_PRIORITY, input.ticket, input.level) },
            sign = { r, sig, id -> TriageCommand.SetPriority(r.tool, sig, id, r.ticket, r.level) },
        ),
    )
}

File: agent-driven-architecture-main/examples/typescript/test/laws/fixtures/quickstart/walk/src/blocks/notes/tools.ts
// ── blocks/notes/tools — APPEND 3: the Verb entry ─────────────────────────
// Name, description, schema, a PURE `run`, and the `sign` that turns the result
// into the command the bus commits. `reversible` because a note is not an
// irreversible act; an irreversible verb would declare `requestedBy` and the
// boundary gate would demand a different principal's confirmation.

import type { Verb } from "@adr/spine/pure/verb";
import { reversible } from "@adr/spine/pure/verb";
import { object, string } from "valibot";
import type { AddNoteCommand, AddNoteResult } from "./contract";

export function notesVerbs<S>(): readonly Verb<S>[] {
  return [
    reversible<S, { text: string }, AddNoteResult, AddNoteCommand>({
      name: "addNote",
      describe: "Record a note on the session.",
      schema: object({ text: string() }),
      run: (input) => ({ outcome: "ok", tool: "addNote", text: input.text }),
      sign: (result, sig, id) => ({
        outcome: "ok",
        tool: "addNote",
        sig,
        id,
        text: result.text,
      }),
    }),
  ];
}
